src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -4251,6 +4251,28 @@ public void Fill_InvalidStartIndexCount_ThrowsArgumentOutOfRangeException(int ar
             AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => Array.Fill(new string[arrayLength], "", startIndex, count));
         }
 
+        public class Bar
+        {
+        }
+
+        public class Foo
+        {
+        }
+
+        [Fact]
+        public static void Fill_Casting()
+        {
+            Bar[] barArray = new Bar[2];
+            Array.Fill<object>(barArray, new Bar());
```

Comment:
Please validate that `Array.Fill` actually populated the array. Something like:

```cs
public class Bar
{
    public string Value;
}

Bar[] barArray = new Bar[]
{
    new Bar() { Value = "1" },
    new Bar() { Value = "2" },
};

Array.Fill<object>(barArray, new Bar() { Value = "x" });

Assert.Equal("x", barArray[0].Value);
Assert.Equal("x", barArray[1].Value);
```

This method should also test the other `Fill` overload. For instance: `new Bar[5]`, with `Fill(theArray, 1, 2)`, ensuring that only _theArray\[1\]_ and _theArray\[2\]_ were changed. There's no need to unit test that the bounds check is correct; I think there's already a unit test for that.

And we should have a test between `uint[]` <-> `int[]` as well, to ensure that we didn't inadvertently break that logic. For instance:

```cs
int[] ints = new int[] { 10, 20, 30, 40, 50 };
uint[] uints = (uint[])(object)ints;
// do work via Array.Fill(uints, ...);
Assert.Equal(new int[] { /* whatever the expected sequence is */ }, ints);
```